inception-workload-dynamic/src/main/java/de/tudarmstadt/ukp/inception/workload/dynamic/page/MonitoringPage.java

```diff
@@ -0,0 +1,456 @@
+        error("No Project selected, please enter the monitoring page only with a valid project reference");
+        setResponsePage(getApplication().getHomePage());
+
+    }
+
+
+
+    //Constructor with a project
+    public MonitoringPage(final PageParameters aPageParameters)
+    {
+        super(aPageParameters);
+
+        //top form initialize
+        workload = new MonitoringForm("workload");
+        workload.setOutputMarkupId(true);
+
+        //Get current Project
+        StringValue projectParameter = aPageParameters.get(PAGE_PARAM_PROJECT_ID);
+        if (getProjectFromParameters(projectParameter).isPresent())
+        {
+            currentProject.setObject(getProjectFromParameters(projectParameter).get());
+        } else {
+            currentProject = null;
+        }
+
+
+        //Get the current user
+        User user = userRepository.getCurrentUser();
+        //Check if Project exists
+        if (currentProject != null)
+        {
+
+
+
+
+            //Check if user is allowed to see the monitoring page of this project
+            if (currentProject != null &&
+                !(projectService.isCurator(currentProject.getObject(), user)
+                || projectService.isManager(currentProject.getObject(), user)))
+            {
+                //Required even in case of error due to wicket
+                add(new ModalWindow("modalWindow"));
+                add(new NumberTextField("defaultDocumentsNumberTextField"));
+                add(new EmptyPanel("dataTable"));
+                error("You have no permission to access project [" + currentProject.getObject().getId() + "]");
+                setResponsePage(getApplication().getHomePage());
+
+            } else {
+                initialize();
+            }
+
+        } else {
+            //Required even in case of error due to wicket
+            add(new ModalWindow("modalWindow"));
+            add(new NumberTextField("defaultDocumentsNumberTextField"));
+            add(new EmptyPanel("dataTable"));
+            //Project does not exists, returning to homepage
+            error("Project [" + projectParameter + "] does not exist");
+            setResponsePage(getApplication().getHomePage());
+
+        }
+
+    }
+
+    public void initialize()
+    {
+
+        //Header of the page
+        Label name = new Label("name",currentProject.getObject().getName());
+        add(name);
+
+        //get Documents for current project
+        documentList.setObject(documentService.
+            listSourceDocuments(currentProject.getObject()));
+
+        //Headers of the table
+        List<String> headers = new ArrayList<>();
+        headers.add(getString("Document"));
+        headers.add(getString("Finished"));
+        headers.add(getString("Processed"));
+        headers.add(getString("Metadata"));
+
+        //Data Provider for the table
+        dataProvider = new DataProvider(documentList.getObject(),
```

Comment:
I think you don't need the `documentList` at all as a instance variable, but can pass the info directly to your DataProvider `dataProvider = new DataProvider(documentService.
                listSourceDocuments(currentProject.getObject()),
            headers, documentService.
            listAnnotationDocuments(currentProject.getObject()));`